pkg/secrets/check_rights_nix.go

```diff
@@ -13,15 +13,23 @@ import (
 	"syscall"
 )
 
-func checkRights(path string) error {
+func checkRights(path string, options checkRightOptions) error {
 	var stat syscall.Stat_t
 	if err := syscall.Stat(path, &stat); err != nil {
 		return fmt.Errorf("invalid executable '%s': can't stat it: %s", path, err)
 	}
 
 	// checking that group and others don't have any rights
-	if stat.Mode&(syscall.S_IRWXG|syscall.S_IRWXO) != 0 {
-		return fmt.Errorf("invalid executable '%s', 'groups' or 'others' have rights on it", path)
+	var unwantedPermissions uint = syscall.S_IRWXO
+	if options.AllowGroupExec {
+		// group should have exec perm
+		unwantedPermissions |= syscall.S_IWGRP
+	} else {
+		// group don't have any rights
+		unwantedPermissions |= syscall.S_IRWXG
+	}
+	if uint(stat.Mode)&(unwantedPermissions) != 0 {
```

Comment:
Is the `uint` cast needed ?

Reply:
It was to be able to compare with `var unwantedPermissions uint = syscall.S_IRWXO`

Reply:
any way I will remove the usage of `unwantedPermissions ` by having 2 different error messages.